src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -1046,31 +986,6 @@ class IMain(@BeanProperty val factory: ScriptEngineFactory, initialSettings: Set
     override def toString = "Request(line=%s, %s trees)".format(line, trees.size)
   }
 
-  def createBindings: Bindings = new IBindings {
-    override def put(name: String, value: Object): Object = {
-      val n = name.indexOf(":")
-      val p: NamedParam = if (n < 0) (name, value) else {
-        val nme = name.substring(0, n).trim
-        val tpe = name.substring(n + 1).trim
-        NamedParamClass(nme, tpe, value)
-      }
-      if (!p.name.startsWith("javax.script")) bind(p)
```

Comment:
Also, this line was intended to avoid the binding by jrunscript of variables with names invalid in scala ("javax.script.x"). Hence it would be interesting to test the new engine with jrunscript, as the problem may still exist.